src/porepy/numerics/nonlinear/nonlinear_solvers.py

```diff
@@ -168,15 +161,13 @@ def newton_step() -> None:
                         break
                     elif is_converged:
                         solver_progressbar.close()
-                        model.after_nonlinear_convergence(
-                            iteration_counter=iteration_counter
-                        )
+                        model.after_nonlinear_convergence()
                         break
 
         if not is_converged:
             model.after_nonlinear_failure()
 
-        return is_converged, iteration_counter
+        return is_converged, model.nonlinear_solver_statistics.num_iteration
```

Comment:
Is it important to return the counter?

Reply:
The only uses of ``NewtonSolver.solve`` in PorePy are in ``run_models.py``, which
ignores both return values. We can freely choose what to return. Although, for a
function that is called solve, having some return value other than ``None`` is
probably good practice.

Reply:
Yes, I'd keep is_converged. But I think we should remove the counter, which can be retreived from the model's solution statistic object.